src/frontend/packages/lib_classroom/src/components/ClassroomWidgetProvider/widgets/RetentionDate/index.tsx

```diff
@@ -68,15 +69,33 @@ export const RetentionDate = () => {
     },
   );
 
-  function onChange(new_retention_date: string | string[]) {
-    let new_retention_date_formatted = null;
-    if (new_retention_date && typeof new_retention_date === 'string') {
-      const utcDateTime = DateTime.fromISO(new_retention_date, { zone: 'utc' });
-      const localDateTime = utcDateTime.toLocal();
-      new_retention_date_formatted = localDateTime.toFormat('yyyy-MM-dd');
+  /**
+   * @param new_retention_date in the format YYYY-MM-DD HH:MM:SS (ISO 8601) UTC
+   * @returns
+   */
+  function onChange(new_retention_date: string | null) {
+    /**
+     * The date is in UTC format so not as the client has chosen.
+     * We need to convert it to local date, then reconvert to YYYY-MM-DD
+     * locale `en-CA` is used to convert the date in the format YYYY-MM-DD
+     */
+    const local_new_retention_date = new_retention_date
+      ? new Date(new_retention_date).toLocaleDateString('en-CA')
```

Comment:
Why using `en-CA`. We never use this locale in the application. Why not `en-US` ?

Reply:
The api want this pattern `YYYY-MM-DD`, so it is a trick to convert to the correct pattern.
`en-US` has this pattern `DD/MM/YYYY`, while `en-CA` has this one `YYYY-MM-DD`.

Anyway I had a problem during my tests with this way, so I changed it in the next commit:
https://github.com/openfun/marsha/pull/2359/commits/8be954b1802619b0b192a2680e010800139639d8#diff-cf30b5417dce67ce54bc983a20ab8a5e92150251f67c5f0209d08e351ac96ab8R68-R72

Reply:
In that case you should not rely on a locale but use the `toFormat` of luxon https://moment.github.io/luxon/#/formatting?id=toformat